homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -179,11 +242,17 @@ def name(self):
 
     async def async_close_cover(self, **kwargs):
         """Close the cover."""
-        await self._async_move(0)
+        current_hass_position = self.current_cover_position
+        steps_to_move = current_hass_position
+        self._async_schedule_update_for_transition(steps_to_move)
+        self._async_update_from_command(await self._shade.close())
 
     async def async_open_cover(self, **kwargs):
         """Open the cover."""
-        await self._async_move(100)
+        current_hass_position = self.current_cover_position
+        steps_to_move = 100 - current_hass_position
+        self._async_schedule_update_for_transition(steps_to_move)
```

Comment:
Same comment as https://github.com/home-assistant/core/pull/62788/files#r868013131 about making this a bit more DRY

Reply:
implemented